pkg/metrics/chaos_daemon.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+)
+
+// DefaultChaosDaemonMetricsCollector is the default metrics collector for chaos daemon
+var DefaultChaosDaemonMetricsCollector = NewChaosDaemonMetricsCollector()
+
+type ChaosDaemonMetricsCollector struct {
+	backgroundProcessManager  bpm.BackgroundProcessManager
+	bpmControlledProcessTotal prometheus.Counter
+	bpmControlledProcesses    prometheus.Gauge
+}
+
+// NewChaosDaemonMetricsCollector initializes metrics for each chaos daemon
+func NewChaosDaemonMetricsCollector() *ChaosDaemonMetricsCollector {
+	return &ChaosDaemonMetricsCollector{
+		backgroundProcessManager: bpm.NewBackgroundProcessManager(nil),
```

Comment:
As the former comments say, you need to use another way to assembly this `ChaosDaemonMetricsCollector` with the certain in-use `bpm` instance. :D

Reply:
OK, I got it.